### PR TITLE
16989: Add bicep sample code

### PR DESCRIPTION
**Pull request by khushal08**
# Overview/Summary

Sample bicep code to run different modules. 

## This PR fixes/adds/changes/removes

1. Provides an example using bicep to run a particular module. 

### Breaking Changes

1. None

## Testing Evidence

1. Tested locally by deploying the code. 
2. Tested PSRule to make sure that the pre-deployment checks pass. 

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Ensured the resource API versions in `.bicep` file/s I am adding/editing are using the latest API version possible
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [x] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [x] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

**Comment by khushal08** [comment]
@jtracey93 Can you please review? thanks

**Review by jtracey93 — CHANGES_REQUESTED** [review]
Hey @khushal08,

Looks like some issues we either need to fix in this PR or create separate issues for to then fix post merging this PR: https://github.com/Azure/ALZ-Bicep/actions/runs/3317756169

Let me know which way you want to go as I this is part of the feature work you have assigned.

Cheers

Jack

**Comment by jtracey93** [comment]
still active

**Comment by jtracey93** [comment]
Still active

File: .ps-rule/DiagLogForAutomation.Rule.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Suppression and rules for unsupported scenarios.
+#
+
+# NOTE:
+# For details on authoring suppression groups see:
+# https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_SuppressionGroups/
+# https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Expressions/
+
+---
+# Synopsis: Ignore automation account audit diagnostic logs are enabled.
```
Comment (jtracey93) [inline .ps-rule/DiagLogForAutomation.Rule.yaml:11]:
```suggestion
# Synopsis: Ignore automation account audit diagnostic logs are enabled as these are covered by DINE policies in ALZ
```

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: .ps-rule/ManagedIdentityForAutomation.Rule.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Suppression and rules for unsupported scenarios.
+#
+
+# NOTE:
+# For details on authoring suppression groups see:
+# https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_SuppressionGroups/
+# https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Expressions/
+
+---
+# Synopsis: Ignore Managed Identity is used for authentication.
```
[the remark this answers is not in the document]
Comment (khushal08) [reply .ps-rule/ManagedIdentityForAutomation.Rule.yaml:11]: I think that's a good point. We need to log a new feature request. Let me open one.

[the remark this answers is not in the document]
Comment (khushal08) [reply .ps-rule/ManagedIdentityForAutomation.Rule.yaml:11]: Created a user story: https://dev.azure.com/CSUSolEng/Azure%20Landing%20Zones/_workitems/edit/25134
My thoughts are for now let's suppress the rule. As part of the fix for the above story, managed identity should be implemented, and suppression rule removed.

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**